### ISSUE TITLE
FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated

### ISSUE THREAD

**Issue by apiszcz** [h5py/h5py#961]
python 3.5.4
>>> h5py.__version__
'2.7.1'
>>> np.__version__
'1.14.0rc1'

```
lib\h5py\__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
```

**Comment by aragilar** [comment]
Will be closed by https://github.com/h5py/h5py/pull/926.

**Comment by arbellea** [comment]
Also get the same error

**Comment by Juanlu001** [comment]
To people coming here from Google: the change was introduced by NumPy 1.14

https://github.com/numpy/numpy/pull/9505

**Comment by bicepjai** [comment]
how to disable this warning none of the options given [here](https://stackoverflow.com/questions/48340392/futurewarning-conversion-of-the-second-argument-of-issubdtype-from-float-to) worked

**Comment by cancan101** [comment]
Any chance to get a new release of h5py?

**Comment by gregory-halverson** [comment]
The warning can be suppressed during import, but it's messy to have to wrap the `h5py` import this way.
```
import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)
import h5py
warnings.resetwarnings()
```

**Comment by babbush** [comment]
I develop an open source project (OpenFermion: https://github.com/quantumlib/OpenFermion) for compiling quantum computing algorithms. We use h5py and so whenever somebody imports openfermion they get this error from h5py. I've had enough people personally email me to complain about this that we're contemplating moving to a different dependency if this is not resolved soon.

**Comment by cancan101** [comment]
It does look like an rc was recently cut: https://github.com/h5py/h5py/releases/tag/2.8.0rc1

**Comment by babbush** [comment]
Ah okay, thanks. I suppose we can update our requirements to insist on at least this version of h5py and that should take care of things.

**Comment by julben** [comment]
Is there any way to avoid having this warning displayed?

**Comment by jakirkham** [comment]
Try the RC?

**Comment by allexlima** [comment]
The `.whl` file of RC is available here -> https://pypi.python.org/pypi/h5py/2.8.0rc1

**Comment by thoppe** [comment]
To those coming in from a google search, this installs the fix

    pip install h5py==2.8.0rc1 

Once this is merged into a standard release, you should be able to `pip install h5py --upgrade`.

**Comment by jeancarlos2020** [comment]
pip install numpy==1.13.3

**Comment by boeddeker** [comment]
For those, where `conda update h5py` currently does not work.
On https://anaconda.org/search?q=h5py is denoted, that currently the default anaconda package has still 2.7.1 and for example, conda-forge has 2.8.0:
```bash
$ conda install -c conda-forge h5py
```

**Comment by RAGHURAM-SADINENI** [comment]
Try installing h5py updated package using the command.
pip install h5py==2.8.0rc1

**Comment by whyboris** [comment]
Finally `2.8.0` is out: https://github.com/h5py/h5py/releases/tag/2.8.0
Just add `h5py==2.8.0` to your `requirements.txt` 🎉

**Comment by ahmedhamdy90** [comment]
To solve this warning for Anaconda now you can use `conda install h5py==2.8.0`

**Comment by maulikmadhavi** [comment]
pip install h5py==2.8.0

**Comment by ksangeek** [comment]
This fixed the problem for me -
```
==> conda install h5py
Solving environment: done

## Package Plan ##

  environment location: /anaconda3

  added / updated specs:
    - h5py


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    conda-4.5.9                |           py36_0         1.0 MB

The following packages will be UPDATED:

    conda: 4.5.4-py36_0 --> 4.5.9-py36_0

Proceed ([y]/n)? y


Downloading and Extracting Packages
conda-4.5.9          |  1.0 MB | ############################################ | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
```

**Comment by KingfaLuis** [comment]
g:\ProgramData\Anaconda3\lib\site-packages\h5py\__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters

**Comment by arfahkT** [comment]
pip install h5py==2.8.0rc1

**Comment by packmanproject** [comment]
at terminal, please  write ( (in this case, I use ubuntu and I have installed h5py module) :

sudo pip install --upgrade h5py (if you use pyhton2)
sudo pip3 install --upgrade h5py (if you use pyhton3)

**Comment by tacaswell** [comment]
I strongly recommend against doing `sudo pip ...` as it will work as you expect only sometimes and will cause problems with your system package manger.

**Comment by jamessong** [comment]
> Try installing h5py updated package using the command.
> pip install h5py==2.8.0rc1

please don't mix conda with pip ... it will be a mess! In fact, why don't they disable the pip from conda?

**Comment by jamessong** [comment]
I am totally confused by the conda junk ... 

So if I update h5py - then a lot of junk messages came up on the screen saying some other packages will be downgraded. So does that mean if I ever run conda update -all, this h5py will break?

Why in the world there is sth. called conda? Created a lot of junk messages on the internet for the lack of proper organization?
